daemon/src/monitor.rs

```diff
@@ -337,11 +344,13 @@ impl Actor<bdk::electrum_client::Client> {
                 tracing::info!("Commit transaction published on chain: {}", txid);
             }
 
-            if let Some(cet) = cet {
-                // Double question mark OK because if we are in PendingCet we must have been
-                // Ready before
-                let txid = broadcast_transaction(&self.client, cet)?;
-                tracing::info!("CET published on chain: {}", txid);
+            if !monitor_cet_timelock {
```

Comment:
Your bug is on line 224. The variable is even called `timelocked_cet` :upside_down_face: 

Just don't set `self.cet` there and you should be fine!

Reply:
why do we set it in `src/model/cfd.rs/ and not here. 

`timelocked_cet` is set below. I didnt change that.

Reply:
> why do we set it in `src/model/cfd.rs/ and not here. 

It is unfortunately a bit tricky but the gist is:

- When we get the attestation, we need to decrypt the CET in any case, even if the timelock is still active. We can't wait.
- The only way to store data in event sourcing is to emit an event, hence we include it in the event.
- Re-hydration of the aggregate will pick up that event and can pull out the data.
- When we later get told that the timelock expired, we can check whether we still have a timelocked CET pending and emit it then. Otherwise we don't have to do anything.

That is why we need to record it in the model CFD's apply method.

> `timelocked_cet` is set below. I didnt change that.

 But that was introduced by your previous PR right? Prior to rebroascasting transactions, we didn't store any of these transactions in this aggregate.

Reply:
You can remove this check now. `self.cet` should only be set to CETs that can be broadcasted!